sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -638,10 +638,16 @@ class UDFSuite extends QueryTest with SharedSparkSession {
   }
 
   test("SPARK-28521 error message for CAST(parameter types contains DataType)") {
-    val e = intercept[AnalysisException] {
-      spark.sql("SELECT CAST(1)")
-    }
-    assert(e.getMessage.contains("Invalid arguments for function cast"))
+    checkError(
+      exception = intercept[AnalysisException] {
+        sql("SELECT CAST(1)")
```

Comment:
https://github.com/apache/spark/blob/37453ad5c85122d11b436813b4f5eddf615a6cf8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L137-L144

In this scenario, the `validParametersCount` is also empty, so `WRONG_NUM_ARGS ` cannot be reused now

Reply:
Do you have any suggestions on the calculation way of `validParametersCount `?